Documentation/Signoffs/Microcontroller.md

```diff
@@ -0,0 +1,46 @@
+
+Within this subsystem, the code will instruct the hardware in how to sample from the sensors, send data to the Communication subsystem, store data onto a USB Flash drive, and monitor battery voltage levels for the user.  These are the core functions that are desired from the microcontroller.  The microcontroller must be a reasonable portion of the end-user's budget of $300.
+
+## CONSTRAINTS ON THE SUBSYSTEM
+The constraints given below are generalized to allow for comparison of any potential contender for the microcontroller.  Constraints on the subsystem are as follows:
+1. The chosen microcontroller should cost no more than $60 as the end-user only has a budget of $300.  This board value cap is gathered from estimating the costs of other subsystems and viewing prices of available options such as Raspberry Pis, Arduinos, Intel, ESP, etc.  Up to 20% of the budget was then allocated due to the importance of this subsystem.
+2. The microcontroller must be able to send instructions and transfer data to other components such as the Communication subsystem.  To do this, Serial Peripheral Interface (SPI) or Universal Asynchronous Receiver/Transmitter (UART) as standards in hardware communication for liaising will need to be achievable. 
+3. USB Flash Drive external storage support is required as per the shall statements. The USB standard is used for convenience from the end user as it is small, easily portable, and very commonly used for data transfer.
+4. The board must have at least 15 digital input pins and 15 analog input pins. A major goal of the project is to allow for synchronous operations of at least 3 air quality sensors.  Sensors may be either analog or digital, so there must be support for both.  The board must have enough Input/Output (I/O) pins to operate any combination of 3 sensors whether they be digital or analog. Different sensors interface with a potentially different number of pins.  Preliminary research shows that a sensor may take up to 5 pins to operate, so 5 pins times 3 sensors gives 15 pins. Other sensors may have more, but it is more common for 5 and under.  
+5. There must be sufficient memory for the code. Comparing similar operating code (Wireless Sensor Collection and Small-time Process Operating) with libraries and compilation show a range of 15 KB (10%) (Kilobytes) above or below 150 KB in flash memory.  To account for the worst case, 165 KB is the required memory.
+6. Some form of battery monitoring capabilities need to be included or applicable as per shall statements.
```

Comment:
Battery monitoring is not always a small/simple task. I'm surprised this is not a separate subsystem.

Reply:
Battery monitoring is now a conjoined effort from the Microcontroller and Power subsystem.